### PR TITLE
Client Generator: Mercure support and various improvements

### PR DESCRIPTION
**Pull request by dunglas**
Includes screenshots and a gif!

See api-platform/client-generator#94,  api-platform/core#2282 and api-platform/demo#66.

### CODE REVIEW

File: client-generator/react.md [hunk 1]
```diff
@@ -1,83 +1,113 @@
-# React generator
+# React Generator
 
-Create a React application using [Facebook's Create React App](https://github.com/facebookincubator/create-react-app):
+![List screenshot](images/react/list.png)
 
-    $ create-react-app my-app
-    $ cd my-app
+The React Client Generator generate a Progressive Web App built with battle-tested libraries from the ecosystem:
```
Comment (bendavies) [inline client-generator/react.md]:
```suggestion
The React Client Generator generates a Progressive Web App built with battle-tested libraries from the ecosystem:
```

File: client-generator/react.md [hunk 2]
```diff
@@ -1,83 +1,113 @@
...
-  </head>
-  <!-- ... -->
+```bash
+$ docker-compose exec client generate-api-platform-client --resource book
 ```
 
-In the app directory, generate the files for the resource you want:
+Omit the resource flag to generate files for all resource types exposed by the API.
+
+If you don't use a standalone installation, run the following command instead:
```
Comment (bendavies) [inline client-generator/react.md]:
```suggestion
If you don't use the standalone installation, run the following command instead:
```

File: client-generator/index.md [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
-* The generated HTML is compatible with [Bootstrap](https://getbootstrap.com/) and includes mandatory classes
-* The generated HTML code is accessible to people with disabilities ([ARIA](https://www.w3.org/WAI/intro/aria) support)
-* The Redux and the React Router configuration is also generated
+* Generate high-quality ES6:
+  * list view (with pagination)
+  * detail view
+  * creation form
+  * edition form
+  * delete button
+* Support to-one and to-many relations
```
Comment (bendavies) [inline client-generator/index.md]:
```suggestion
* Supports to-one and to-many relations
```

File: client-generator/index.md [hunk 4]
```diff
@@ -1,24 +1,32 @@
...
+* Use the appropriate input type (`number`, `date`...)
```
Comment (bendavies) [inline client-generator/index.md]:
```suggestion
* Uses the appropriate input type (`number`, `date`...)
```

File: client-generator/index.md [hunk 5]
```diff
@@ -1,24 +1,32 @@
...
+* Client-side validation
+* Subscribe to data updates pushed by server supporting [the Mercure protocol](https://mercure.rocks)
```
Comment (bendavies) [inline client-generator/index.md]:
```suggestion
* Subscribe to data updates pushed by servers supporting [the Mercure protocol](https://mercure.rocks)
```

File: client-generator/react-native.md [hunk 6]
```diff
@@ -1,40 +1,44 @@
-React Native generator
-======================
+# React Native generator
 
-Create a React Native application using [React Community's Create React Native App](https://github.com/react-community/create-react-native-app)
+![List](images/list.png) 
+
+## Install
+
+To use this generator you need a proper [Node.js](https://nodejs.org/) and [Yarn](https://yarnpkg.com/) (or NPM) installation.
```
Comment (bendavies) [inline client-generator/react-native.md]:
```suggestion
To use this generator you need [Node.js](https://nodejs.org/) and [Yarn](https://yarnpkg.com/) (or [YPM](https://www.npmjs.com/)) installed.
```

File: client-generator/react-native.md [hunk 7]
```diff
@@ -1,40 +1,44 @@
...
+
+In the app directory, generate the files for the resource you want:
 
 ```bash
-$ yarn global add @api-platform/client-generator
+$ npx @api-platform/client-generator https://demo.api-platform.com . -g react-native --resource book
 ```
 
-In the app directory, generate the files for the resource you want:
+Replace the URL by the entrypoint of your Hydra-enabled API.
```
Comment (bendavies) [inline client-generator/react-native.md]:
```suggestion
Replace the URL with the entrypoint of your Hydra-enabled API.
```

File: client-generator/react-native.md [hunk 8]
```diff
@@ -49,16 +53,31 @@ const RouterComponent = () => {
 
 export default RouterComponent;
 ```
-Here is an example of  **App.js**
+
+Here is an example of a `App.js` file:
```
Comment (bendavies) [inline client-generator/react-native.md]:
```suggestion
Here is an example of an `App.js` file:
```

File: client-generator/react.md [hunk 9]
```diff
@@ -1,83 +1,113 @@
...
-    $ yarn global add @api-platform/client-generator
+The easiest way to get started is to install [the API Platform distribution](../distribution/index.md).
+It contains the React Client Generator, all dependencies it needs, a Progressive Web App skeleton generated with Create React App,
+a development Docker container to serve the webapp, and all the API Platform components you may need, including an API server
+supporting Hydra.
 
-Reference the Bootstrap CSS stylesheet in `public/index.html` (optional):
+If you use the API Platform, jump to the next section!
+Alternatively, you can generate a skeleton and install the generator using [npx](https://www.npmjs.com/package/npx).
+To run it, you need a proper [Node.js](https://nodejs.org/) and [Yarn](https://yarnpkg.com/) (or NPM) installation.
```
Comment (bendavies) [inline client-generator/react.md]:
```suggestion
To use this generator you need [Node.js](https://nodejs.org/) and [Yarn](https://yarnpkg.com/) (or [YPM](https://www.npmjs.com/)) installed.
```

File: client-generator/react-native.md [hunk 10]
```diff
@@ -1,40 +1,44 @@
...
 ```bash
-$ yarn add -g expo-cli
+$ yarn global add expo-cli
 $ expo init my-app
+# When asked, choose to use the blank template
 $ cd my-app
 ```
 
-Install:  React Native Elements, React Native Router Flux, React Native Vector Icons, Redux, React Redux, Redux Thunk, Redux Form, Prop Types
+Install the requires dependencies:
```
Comment (antograssiot) [inline client-generator/react-native.md]: required ?

**Pull request merged**